src/Facebook/Url/FacebookUrlDetectionHandler.php

```diff
@@ -160,4 +160,23 @@ protected function getHeader($key)
     {
         return $this->getServerVar('HTTP_' . $key);
     }
+
+    /**
+     * Checks if the value in X_FORWARDED_HOST is a valid hostname
+     * Could prevent unintended redirections
+     *
+     * @param string $header
+     *
+     * @return boolean
+     */
+    protected function isValidForwardedHost($header)
+    {
+        $elements = explode(',', $header);
+        $host = $elements[count($elements) - 1];
+        
+        return preg_match("/^([a-z\d](-*[a-z\d])*)(\.([a-z\d](-*[a-z\d])*))*$/i", $host) //valid chars check
```

Comment:
the regex is not equivalent either. The current regex does not allow to have a dash followed by a dot, while your regex allows it @yguedidi